### PR TITLE
refactor: add some utils to generate spendable cells

### PR DESCRIPTION
**Pull request by keroro520**

### CODE REVIEW

**Review by chuijiaolianying — APPROVED** [review]

File: test/src/specs/relay/transaction_relay_low_fee_rate.rs [hunk 1]
```diff
@@ -1,95 +1,47 @@
...
-            .as_builder()
-            .capacity(capacity.pack())
-            .build();
-        let tx_low_fee = tx_low_fee
-            .data()
-            .as_advanced_builder()
-            .set_outputs(vec![])
-            .output(output)
+        let cells = gen_spendable(node0, 1);
+        // As for `low_fee`, which is `inputs.total_capacity == outputs.total_capacity`,
```
Comment (chuijiaolianying) [inline test/src/specs/relay/transaction_relay_low_fee_rate.rs:21]: `inputs.total_capacity == outputs.total_capacity` more like zero_fee, but of course it also make sense in this case.

File: test/src/specs/rpc/truncate.rs [hunk 2]
```diff
@@ -1,24 +1,24 @@
...
 impl Spec for RpcTruncate {
     // After truncating, the chain will be rollback to the target block, and tx-pool be cleared.
     fn run(&self, nodes: &mut Vec<Node>) {
         let node = &nodes[0];
-        node.generate_blocks(12);
+
+        let cells = gen_spendable(node, 2);
+        let transactions = always_success_transactions(node, &cells);
+        let tx1 = &transactions[0];
+        let _tx2 = &transactions[1];
```
Comment (chuijiaolianying) [inline test/src/specs/rpc/truncate.rs:15]:
```suggestion
        let tx2 = &transactions[1];
```

File: test/src/specs/rpc/truncate.rs [hunk 3]
```diff
@@ -1,24 +1,24 @@
...
-            tx1
-        };
-        node.generate_blocks(3);
-        let _tx2 = {
-            let tx2 = node.new_transaction_spend_tip_cellbase();
-            node.submit_transaction(&tx2);
-            tx2
-        };
+
+        node.submit_transaction(&transactions[0]);
```
Comment (chuijiaolianying) [inline test/src/specs/rpc/truncate.rs:19]:
```suggestion
        node.submit_transaction(tx1);
```

File: test/src/specs/rpc/truncate.rs [hunk 4]
```diff
@@ -1,24 +1,24 @@
...
+        node.generate_blocks(DEFAULT_TX_PROPOSAL_WINDOW.0 as usize + 1);
+        node.submit_transaction(&transactions[1]);
```
Comment (chuijiaolianying) [inline test/src/specs/rpc/truncate.rs:21]:
```suggestion
        node.submit_transaction(tx2);
```

**Comment by doitian** [comment]
@keroro520 many conflicts

**Comment by keroro520** [comment]
I am going to resolve conflicts and make some modifications based on review suggestions.

**Comment by keroro520** [comment]
Resolved conflicts

File: test/src/util/cell.rs [hunk 5]
```diff
@@ -0,0 +1,118 @@
...
+use crate::Node;
+use ckb_types::core::cell::{CellMeta, CellMetaBuilder};
+use ckb_types::core::{BlockView, TransactionInfo};
+use ckb_types::packed::{Byte32, CellInput, CellOutput, OutPoint};
+use ckb_types::prelude::*;
+use std::collections::HashMap;
+use std::ops::Range;
+
+pub fn gen_spendable(node: &Node, count: usize) -> Vec<CellMeta> {
+    // TODO optimize based on `out_of_boostrap_period`
```
Comment (chuijiaolianying) [inline test/src/util/cell.rs:11]: `out_bootstrap_period` is already in develop branch

**Review by chuijiaolianying — APPROVED** [review]

**Review by yangby-cryptape — APPROVED** [review]

**Comment by doitian** [comment]
bors r=chuijiaolianying,yangby-cryptape

**Review by doitian — APPROVED** [review]

**Pull request merged**